### PR TITLE
Add file BBEs

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

### CODE REVIEW

File: new-examples/directories/directories.bal [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    // Checks whether the directory of the provided path exists.
+    boolean fileExists = check file:test("foo", file:EXISTS);
+    io:println("Is foo directory exist: ", fileExists.toString());
+
+    // Copies the directory to the new path.
+    check file:copy(dirPath, "test", file:REPLACE_EXISTING);
+    io:println("The " + dirPath + " directory copied successfully.");
+
+    // Renames the directory to the new path.
```
Comment (praneesha) [inline new-examples/directories/directories.bal]:
```suggestion
    // Renames the directory in the new path.
```

File: new-examples/directories/directories.metatags [hunk 2]
```diff
@@ -0,0 +1,2 @@
+description: This BBE shows how to perform directory operations in Ballerina.
```
Comment (praneesha) [inline new-examples/directories/directories.metatags:1]:
```suggestion
description: This BBE shows how to perform directory operations in Ballerina.
```
Please change this in all BBEs.

File: new-examples/directories/directories.metatags [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+keywords: ballerina, ballerina by examples, bbe, directory, file, path
```
Comment (praneesha) [inline new-examples/directories/directories.metatags:2]:
```suggestion
keywords: ballerina, ballerina by examples, bbe, directory, file, path
```
Please change this in all BBEs.

File: new-examples/filepaths/filepath.bal [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    // Gets the base name of the path.
+    string name = check file:basename("/A/B/C");
+    io:println("Filename of /A/B/C: ", name);
+
+    // Gets the enclosing parent directory.
+    string parentPath = check file:parentPath("/A/B/C");
+    io:println("Parent of /A/B/C: ", parentPath);
+
+    // Gets the shortest path name equivalent to path by purely lexical processing.
```
Comment (praneesha) [inline new-examples/filepaths/filepath.bal:21]:
```suggestion
    // Gets the shortest path name equivalent to the path by purely lexical processing.
```

File: new-examples/filepaths/filepath.metatags [hunk 5]
```diff
@@ -0,0 +1,2 @@
+description: This BBE shows how to manipulate file paths in a way that is compatible with the target Operating System in Ballerina.
```
Comment (praneesha) [inline new-examples/filepaths/filepath.metatags:1]:
```suggestion
description: This BBE shows how to manipulate file paths in a way that is compatible with the target Operating System in Ballerina.
```
Please change.

File: new-examples/filepaths/filepath.metatags [hunk 6]
```diff
@@ -0,0 +1,2 @@
...
+keywords: ballerina, ballerina by examples, bbe, file, directory, path, filepath
```
Comment (praneesha) [inline new-examples/filepaths/filepath.metatags:2]:
```suggestion
keywords: ballerina, ballerina by examples, bbe, file, directory, path, filepath
```
Please change.

File: new-examples/filepaths/tests/filepath_test.bal [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import ballerina/test;
+import ballerina/os;
+
+string[] outputs = [];
+
+// This is the mock function which will replace the real function
```
Comment (praneesha) [inline new-examples/filepaths/tests/filepath_test.bal:6]:
```suggestion
// This is the mock function, which will replace the real function.
```

File: new-examples/filepaths/tests/filepath_test.bal [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+public function mockPrint(any... val) {
+    outputs.push(val.reduce(function (any a, any b) returns string => a.toString() + b.toString(), "").toString());
+}
+
+boolean isWindows = os:getEnv("OS") != "";
+
+@test:Config {}
+function testFunc() returns error? {
+    test:when(mock_printLn).call("mockPrint");
+    // Invoking the main function
```
Comment (praneesha) [inline new-examples/filepaths/tests/filepath_test.bal:22]:
```suggestion
    // Invoking the main function.
```

File: new-examples/files/file.bal [hunk 9]
```diff
@@ -0,0 +1,40 @@
+import ballerina/file;
+import ballerina/io;
+
+public function main() returns error? {
+
+    // Creates a file in the given file path.
+    check file:create("bar.txt");
+    io:println("The bar.txt file created successfully.");
+
+    // Checks whether the file of the provided path exists.
```
Comment (praneesha) [inline new-examples/files/file.bal:10]:
```suggestion
    // Checks whether the file exists on the provided path.
```

File: new-examples/files/file.bal [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    boolean fileExists = check file:test("bar.txt", file:EXISTS);
+    io:println("Is the bar.txt file exist: ", fileExists.toString());
+
+    // Checks whether the file is readable or not.
+    fileExists = check file:test("bar.txt", file:READABLE);
+    io:println("Is the bar.txt file readable: ", fileExists.toString());
+
+    // Checks whether the file is readable or not.
```
Comment (praneesha) [inline new-examples/files/file.bal:18]:
```suggestion
    // Checks whether the file is writable or not.
```

File: new-examples/files/file.bal [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    io:println("Is the bar.txt file writeable: ", fileExists.toString());
+
+    // Copies the file or directory to the new path.
+    check file:copy("bar.txt",  "bar1.txt", file:REPLACE_EXISTING);
+    io:println("The bar.txt file copied successfully.");
+
+    check file:rename("bar.txt", "bar2.txt");
+    io:println("The bar.txt file renamed successfully.");
+
+    // Gets metadata information of the file.
```
Comment (praneesha) [inline new-examples/files/file.bal:29]:
```suggestion
    // Gets the metadata information of the file.
```

File: new-examples/files/file.metatags [hunk 12]
```diff
@@ -0,0 +1,2 @@
+description: This BBE shows how to perform file-system operations in Ballerina.
```
Comment (praneesha) [inline new-examples/files/file.metatags:1]:
```suggestion
description: This BBE shows how to perform file-system operations in Ballerina.
```
Please change.

File: new-examples/files/file.metatags [hunk 13]
```diff
@@ -0,0 +1,2 @@
...
+keywords: ballerina, ballerina by examples, bbe, file, path
```
Comment (praneesha) [inline new-examples/files/file.metatags:2]:
```suggestion
keywords: ballerina, ballerina by examples, bbe, file, path
```
Please change.

File: new-examples/temp-files-directories/temp_files_directories.metatags [hunk 14]
```diff
@@ -0,0 +1,2 @@
+description: This BBE shows how to perform file-system operations in Ballerina.
```
Comment (praneesha) [inline new-examples/temp-files-directories/temp_files_directories.metatags:1]:
```suggestion
description: This BBE shows how to perform file-system operations in Ballerina.
```
Please change.

File: new-examples/temp-files-directories/temp_files_directories.metatags [hunk 15]
```diff
@@ -0,0 +1,2 @@
...
+keywords: ballerina, ballerina by examples, bbe, file, path, temp
```
Comment (praneesha) [inline new-examples/temp-files-directories/temp_files_directories.metatags:2]:
```suggestion
keywords: ballerina, ballerina by examples, bbe, file, path, temp
```
Please change.

File: new-examples/index.json [hunk 16]
```diff
@@ -64,5 +64,28 @@
...
             }
         ]
+    },
+    {
+        "title": "File",
+        "column": 0,
+        "category": "Language concepts",
+        "samples": [
+            {
+                "name": "File Path",
```
Comment (daneshk) [inline new-examples/index.json:74]:
```suggestion
                "name": "File Paths",
```

File: new-examples/directories/directories.metatags (hunk 3 above)
Comment (kalaiyarasiganeshalingam) [reply new-examples/directories/directories.metatags:2]: Contained the related keywords of the BBE but all BBEs are related to the file module. So these may have the same keywords Can you check them?

File: new-examples/index.json [hunk 17]
```diff
@@ -64,5 +64,28 @@
...
+            },
+            {
+                "name": "Files",
+                "url": "files"
+            },
+            {
+                "name": "Temp Files and Directories",
+                "url": "temp-files-directories"
+            }
+        ]
```
Comment (daneshk) [inline new-examples/index.json:89]: Shall we remove these entries. we don't need to update index.json file

File: new-examples/directories/directories.metatags (hunk 2 above)
Comment (kalaiyarasiganeshalingam) [reply new-examples/directories/directories.metatags:1]: Contained the specific description of the BBEs in the description but all BBEs are related to the file module. So these may seem the same description. Can you check them?

**Review by daneshk — APPROVED** [review]
LGTM

File: new-examples/directories/directories.bal [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+
+    // Checks whether the directory of the provided path exists.
+    boolean dirExists = check file:test("foo", file:EXISTS);
+    io:println("Is foo directory exist: ", dirExists.toString());
+
+    // Copies the directory with another name.
+    check file:copy(dirPath, "test", file:REPLACE_EXISTING);
+    io:println("The " + dirPath + " directory copied successfully.");
+
+    // Renames the directory in the new another name.
```
Comment (praneesha) [inline new-examples/directories/directories.bal:19]:
```suggestion
    // Renames the directory to another new name.
```

File: new-examples/directories/directories.metatags (hunk 2 above)
Comment (praneesha) [reply new-examples/directories/directories.metatags:1]: +1

File: new-examples/directories/directories.metatags (hunk 3 above)
Comment (praneesha) [reply new-examples/directories/directories.metatags:2]: +1

File: new-examples/filepaths/tests/filepath_test.bal [hunk 19]
```diff
@@ -0,0 +1,57 @@
+import ballerina/test;
+import ballerina/os;
+
+string[] outputs = [];
+
+// This is the mock function which will replace the real function.
```
Comment (praneesha) [inline new-examples/filepaths/tests/filepath_test.bal:6]:
```suggestion
// This is the mock function, which will replace the real function.
```

**Review by praneesha — APPROVED** [review]

**Pull request merged**